File: hs/Mod_fromBinary.hs
module Mod_fromBinary ( fromBinary ) where
import Prelude hiding (and, or) 
import Debug.Trace
import Splib
import Types

fromBinary bin = fromBinaryHelper bin 0 1

fromBinaryHelper bin res mult 
   |empty bin = res
   |otherwise = fromBinaryHelper (removeLast bin) (res+(last bin)*mult) (mult*2)

File: start/cpp/Mod_fromBinary.cpp
#pragma once

int fromBinary(List<Int> bin){
	int i = 0;
	int mult = 1;
	while(!empty(bin)){
		i += last(bin) * mult;
		mult = mult *2;
		bin = removeLast(bin);
	}
	
	return i;
}